equinix/resource_fabric_connection.go

```diff
@@ -0,0 +1,265 @@
+	_, res, err := client.ConnectionsApi.UpdateConnectionByUuid(ctx, updates, uuid)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf(" Error response for the connection update, response %v, error %v", res, err))
+	}
+	updatedConn := v4.Connection{}
+	updatedConn, err = waitForConnectionUpdateCompletion(uuid, meta, ctx)
+
+	if err != nil {
+		return diag.FromErr(fmt.Errorf(" Errored while waiting for successful connection update, response %v, error %v", res, err))
+	}
+
+	d.SetId(updatedConn.Uuid)
+	return setFabricMap(d, updatedConn)
+}
+
+func waitForConnectionUpdateCompletion(uuid string, meta interface{}, ctx context.Context) (v4.Connection, error) {
+	log.Println("Waiting for connection update to complete")
+	stateConf := &resource.StateChangeConf{
+		Target: []string{"COMPLETED"},
+		Refresh: func() (interface{}, string, error) {
+			client := meta.(*Config).fabricClient
+			dbConn, _, err := client.ConnectionsApi.GetConnectionByUuid(ctx, uuid, nil)
+			if err != nil {
+				return "", "", err
+			}
+			updatableState := ""
+			if "COMPLETED" == dbConn.Change.Status {
+				updatableState = dbConn.Change.Status
+			}
+			return dbConn, updatableState, nil
+		},
+		Timeout:    2 * time.Minute,
+		Delay:      30 * time.Second,
+		MinTimeout: 30 * time.Second,
+	}
+
+	inter, err := stateConf.WaitForStateContext(ctx)
+	dbConn := v4.Connection{}
+
+	if err == nil {
+		dbConn = inter.(v4.Connection)
+	}
+	return dbConn, err
+}
+
+func waitUntilConnectionIsActive(uuid string, meta interface{}, ctx context.Context) (v4.Connection, error) {
+	log.Println("Waiting for connection to be in active state ")
+	stateConf := &resource.StateChangeConf{
+		Target: []string{"ACTIVE"},
+		Refresh: func() (interface{}, string, error) {
+			client := meta.(*Config).fabricClient
+			dbConn, _, err := client.ConnectionsApi.GetConnectionByUuid(ctx, uuid, nil)
+
+			if err != nil {
+				return "", "", err
+			}
+			updatableState := ""
+			if "ACTIVE" == *dbConn.State {
+				updatableState = string(*dbConn.State)
+			}
+			return dbConn, updatableState, nil
+		},
+		Timeout:    5 * time.Minute,
+		Delay:      30 * time.Second,
+		MinTimeout: 30 * time.Second,
+	}
+
+	inter, err := stateConf.WaitForStateContext(ctx)
+	dbConn := v4.Connection{}
+
+	if err == nil {
+		dbConn = inter.(v4.Connection)
+	}
+	return dbConn, err
+}
+
+func resourceFabricConnectionDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	log.Printf(" inside fabric connection delete resource ")
+	diags := diag.Diagnostics{}
+	client := meta.(*Config).fabricClient
+	ctx = context.WithValue(ctx, v4.ContextAccessToken, meta.(*Config).FabricAuthToken)
+	uuid := d.Id()
+	if uuid == "" {
+		return diag.Errorf("No uuid found %v ", uuid)
+	}
+	_, resp, err := client.ConnectionsApi.DeleteConnectionByUuid(ctx, uuid)
+	if err != nil {
+		fmt.Errorf(" Error response for the connection delete error %v and response %v", err, resp)
```

Comment:
I noticed a few `fmt.Errorf(" Error` references in this PR.  Is there any reason for the leading space? I assume not and I suspect the space can and should be removed.

Reply:
Yes, Java Convention I use for any log entry with leading and trailing space so that actual error will not suffixed to the message. If it is an issue with GO, I will remove it.